### DISCUSSION TITLE
Is Stride the right tool for the job and where would be the best place to start.

### DISCUSSION THREAD

**Discussion by Beotheric** [stride3d/stride#2483] [Newcomers]
The "job": A turn based strategy game where players queue up commands to their units, and only once all players have ended their turn does the game acts on them and presents the new state of the world that resulted to the players (the so called simultaneous turn system). 

More specifically what I am looking at doing is having the code that processes the commands and updates the state of the world (the Domain) as its own layer with no dependencies on other parts of the system (i.e. it can be developed independently as it's own class library) and then effectively use Stride to provide the presentation layer that allows players to issue commands and view the world.

I'd like to be able to do this (the Stride bits) via the Game Studio as that seems the easier "introduction", but it feels like this needs to be Code First - as none of the tutorials/help videos I've found tell me how to achieve this via the Game Studio. Any help about where to start (what's the right approach, what tutorials/videos are a good place to begin my learning with this in mind) would be appreciated.

**Comment by VaclavElias** [comment]
I expect you'll get more precise answers from the community, but as an example, the game Distant Worlds 2 was made using Stride. The game developers forked and customized the engine to suit their needs (though we don’t have exact details).

https://store.steampowered.com/news/app/1531540/view/4675388674389181062

**Reply by Doprez** [reply]
Another similar game is [Children of the galaxy](https://store.steampowered.com/app/563390/Children_of_the_Galaxy/) and they have some of their repo public to give an example of separated data from the main game. https://github.com/EmptyKeys/Children-of-the-Galaxy.

As for the code first, it is a C# project at the heart so its a bit easier than its usual state of being the scripting language. Everything is initialized from the [Game](https://doc.stride3d.net/4.2/en/api/Stride.Engine.Game.html) and there are some examples of modifying that class seen in [community examples](https://doc.stride3d.net/4.2/en/community-resources/index.html). Unfortunately code only tutorials are very sparse but @VaclavElias has done a lot of work with the [Stride Community Toolkit](https://github.com/stride3d/stride-community-toolkit/tree/main/examples/code-only) with many code first examples. I also primarily use VS/Rider when developing and only really open GameStudio in order to modifiy the scene and create the map so if thats your workflow it works well also.
